api/dp-openapi.yaml

```diff
@@ -502,6 +502,113 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/codepoints:
+    get:
+      tags:
+      - project
+      description: Returns a string representing valid codepoints for the project
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: Codepoints
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/codepoints'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+  /projects/{projectid}/pages/checkout:
+    post:
+      tags:
+      - project
+      description: Checks out a page to the user
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      - name: roundid
+        in: query
+        description: round ID
+        required: true
+        schema:
+          type: string
+      responses:
+        200:
+          description: page_name and page_state
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/checked_page'
+        '401':
+          $ref: '#/components/responses/Unauthorized'
+        '404':
+          $ref: '#/components/responses/NotFound'
+        '429':
+          $ref: '#/components/responses/RateLimitExceeded'
+        default:
+          $ref: '#/components/responses/UnexpectedError'
+
+  /projects/{projectid}/pages/{pagename}/rounds/{roundid}:
+    put:
+      tags:
+      - project
+      description: Operates on a page checked out to a user
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
+      - name: pagename
+        in: path
+        description: Name of page
+        required: true
+        schema:
+          type: string
+      - name: roundid
+        in: path
+        description: round ID
+        required: true
+        schema:
+          type: string
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
```

Comment:
we can't be more specific here?

Reply:
We probably want to hold off on reviewing the swagger spec until we've worked out the final details on the interface, then we can make sure the swagger spec is up-to-date. But yes, we need to be clear on the schema that is being returned from this call.